### ISSUE TITLE
Feature request: another option to split containers

### ISSUE THREAD

**Issue by chrm** [i3/i3#3238]
A container can be set to either split horizontally or vertically. I'm suggesting a third option that would split a container depending on it's proportions.

If the container is wider than tall -> split horizontally
If the container is taller than wide -> split vertically

I think it would be much easier and faster to get a nice layout this way. Imagine what happens when you have a 16:9 screen and open four windows. If the screen is set to split horizontally you get four thin windows side by side, each taking up 1/4 of the screen. If the screen is set to split by proportion you get one window taking up 1/2 of the screen, one window with 1/4, and two with 1/8.

Also, each time you open a new window, only one other window has to be resized to make room. That helps with keeping the windows you already have open at the right size.

**Comment by Airblader** [comment]
You can achieve this easily by executing `split toggle` whenever a new window opens (perhaps with the exception of the first window on a workspace, depending on your taste). So this can be trivially achieved via the IPC (or even configuration only).

**Comment by Airblader** [comment]
Reopening for now.

**Comment by Airblader** [comment]
My initial thought is that this might be a good addition. However, we need to make sure that the exact direction is calculated on-the-fly, otherwise we'd have to recalculate on all kinds of events (tree operations, RandR changes, …). It also affects displaying the indicator correctly etc.

**Comment by chrm** [comment]
After tinkering with my configuration for a while I just realised, my description wasn't very good. I was thinking about a container that in some cases doesn't exist yet. Let me try again.

The split direction of a workspace already gets set accordingly to it's proportions. That means, if you have only one window on a workspace, the first split will half that window and put the new window on the right (wide workspace) or on the bottom (tall workspace). I'm suggesting an option to make that behavior recursive.

I think it's visually more intuitive than the approach to squish everything in the current container and put the new window at the end.

Here is how I think it could work. Keep in mind, I have not looked at the source code yet.

Lets call it split-recursive.

    if i3 has to create a new window
        take the focused window W and its container C
        if container C is set to split-recursive
            if window W is the only thing in container C
                create the new window W2 in container C, acording to C's proportions
            else
                put a new container C2 around window W
                set C2 to split-recursive
                create the new window W2 in container C2, acording to C2's proportions

I don't think there is any need for an indicator, because you can tell what happens just by the dimensions of the focused window.

**Comment by Airblader** [comment]
This would require us to constantly create new containers on our own. I'm not a fan of that idea, it's not something we really do or want to do.

What about https://github.com/olemartinorg/i3-alternating-layout? That seems to do exactly what you want, no?

**Comment by Airblader** [comment]
Closing this since it can be solved outside of i3, even if not through configuration only, and such tooling already exists.
